python/lbann/modules/base.py

```diff
@@ -216,34 +309,35 @@ def __init__(self, num_dims,
     def forward(self, x):
         self.instance += 1
         name = '{0}_instance{1}'.format(self.name, self.instance)
+
+        convtype = ('_deconv' if self.transpose else '_conv')
+        kwargs = {}
+        kwargs['weights'] = self.weights
+
+        kwargs['name'] = (name+convtype if self.activation else name)
+        kwargs['num_dims'] = self.num_dims
+        kwargs['num_output_channels'] = self.out_channels
+        kwargs['has_bias'] = self.bias
+        kwargs['num_groups'] = self.groups
+        kwargs['parallel_strategy'] = self.parallel_strategy
+        kwargs['has_vectors'] = self.has_vectors
+
+        if (self.has_vectors):
+          kwargs['conv_dims'] = self.kernel_size
+          kwargs['conv_pads'] = self.padding
+          kwargs['conv_dilations'] = self.dilation
+          kwargs['conv_strides'] = self.stride
+        else:
+          kwargs['conv_dims_i'] = self.kernel_size
+          kwargs['conv_pads_i'] = self.padding
+          kwargs['conv_dilations_i'] = self.dilation
+          kwargs['conv_strides_i'] = self.stride
```

Comment:
If we commit to using the `has_vectors` API internally:
```suggestion
        kwargs['has_vectors'] = True
        kwargs['conv_dims'] = str_list(self.kernel_dim)
        kwargs['conv_pads'] = str_ilst(self.padding)
        kwargs['conv_dilations'] = str_list(self.dilation)
        kwargs['conv_strides'] = str_list(self.stride)
```

Reply:
This is a better way to do it. Changed to this in latest commits.